### PR TITLE
fix(swagger-generator): host property generation

### LINKED ISSUE #631: Swagger connector generator generates wrong host connector parameter
In Swagger connector generator we use the `host` parameter from the Swagger specification, we should also consider `schemes` property and generate the URL as the underlying REST Swagger Camel component.

### PR DESCRIPTION
**Pull request by zregvart**
Swagger connector generator now considers `schemes` from Swagger
specification and the Swagger specification URL in order to determine
proper `host` property for the connector being generated.

Fixes  #631

### CODE REVIEW

**Comment by jimmidyson** [comment]
I assume we only support Swagger 2.0? Are we ignoring OpenAPI 3.0 for now?

File: app/rest/connector-generator/src/main/java/io/syndesis/connector/generator/swagger/PropertyGenerators.java [hunk 1]
```diff
@@ -114,6 +131,38 @@
...
+            schemeToUse = "https";
+        } else {
+            schemeToUse = schemes.get(0).toValue();
+        }
+
+        final String host = swagger.getHost();
+        String hostToUse;
+        if (StringUtils.isEmpty(host)) {
+            hostToUse = requireNonNull(specificationUrl, "Swagger specification does not provide a `host` definition "
+                + "and the Swagger specification was uploaded so the originating URL is lost to determine the scheme to use").getHost();
```
Comment (jimmidyson) [inline app/rest/connector-generator/src/main/java/io/syndesis/connector/generator/swagger/PropertyGenerators.java]: `so it is impossible to determine the originating URL` perhaps?

File: app/rest/connector-generator/src/main/java/io/syndesis/connector/generator/swagger/SwaggerConnectorGenerator.java [hunk 2]
```diff
@@ -107,7 +110,11 @@ public ConnectorSummary info(final ConnectorTemplate connectorTemplate, final Co
     /* default */ Connector basicConnector(final ConnectorTemplate connectorTemplate, final ConnectorSettings connectorSettings) {
         final Swagger swagger = parseSpecification(connectorSettings);
 
-        requiredSpecification(connectorSettings);
+        final String specification = requiredSpecification(connectorSettings);
```
Comment (jimmidyson) [inline app/rest/connector-generator/src/main/java/io/syndesis/connector/generator/swagger/SwaggerConnectorGenerator.java]: `specificationURL` perhaps?

**Comment by zregvart** [comment]
> I assume we only support Swagger 2.0? Are we ignoring OpenAPI 3.0 for now?

That is correct, I can't seem to find OpenAPI 3.0 Java implementation that we can use, and also it simplifies things a bit. Also #467

**Comment by zregvart** [comment]
@jimmidyson we should be good to go here, mind having another look?

**Pull request merged**